baseapp/abci.go

```diff
@@ -158,50 +159,94 @@ func (app *BaseApp) EndBlock(req abci.RequestEndBlock) (res abci.ResponseEndBloc
 // the ante handler (which checks signatures/fees/ValidateBasic).
 //
 // NOTE:CheckTx does not run the actual Msg handler function(s).
-func (app *BaseApp) CheckTx(req abci.RequestCheckTx) (res abci.ResponseCheckTx) {
-	var result sdk.Result
-
+func (app *BaseApp) CheckTx(req abci.RequestCheckTx) abci.ResponseCheckTx {
 	tx, err := app.txDecoder(req.Tx)
+	if err != nil {
+		space, code, log := sdkerrors.ABCIInfo(err, false)
```

Comment:
Logic looks good, but I'm sure you will be doing this a few places.
I would recommend adding `CheckTxError(err error) abci.ResponseCheckTx` function (in types? in baseapp?) and then using it these places to make error handling less of a chore

Reply:
Added functions to convert errors to the correct ABCI response types.